### PR TITLE
Enable style manager in CarPlay

### PR DESCRIPTION
**Pull request by 1ec5**
The navigation v4 styles are now visible in CarPlay, both on the main map and in turn-by-turn navigation.

/cc @vincethecoder @bsudekum

### CODE REVIEW

File: MapboxNavigation/DayStyle.swift [hunk 1]
```diff
@@ -233,18 +233,31 @@ open class NightStyle: DayStyle {
...
 
-open class CarPlayDayStyle: DayStyle {
+@available(iOS 12.0, *)
+open class CarPlayDayStyle: DayStyle, CarPlayStyle {
+    public var previewStyleURL = URL(string: "mapbox://styles/mapbox/navigation-preview-day-v4")!
+    
     open override func apply() {
         super.apply()
         ExitView.appearance().foregroundColor = .white
         GenericRouteShield.appearance().foregroundColor = .white
```
Comment (vincethecoder) [inline MapboxNavigation/DayStyle.swift]: @1ec5 `CarPlayDayStyle` should apply: 
```
ExitView.appearance().foregroundColor = .black
GenericRouteShield.appearance().foregroundColor = .black
```

Comment (vincethecoder) [reply MapboxNavigation/DayStyle.swift]: See reference: 
[Black foreground color for Day Styles' ExitView](https://github.com/mapbox/mapbox-navigation-ios/blob/master/MapboxNavigation/DayStyle.swift#L91)
[GenericRouteShield.appearance().foregroundColor = .black](https://github.com/mapbox/mapbox-navigation-ios/blob/master/MapboxNavigation/DayStyle.swift#L97)

Comment (frederoni) [reply MapboxNavigation/DayStyle.swift]: The background is dark on CarPlay, so we need to override the DayStyle to get some contrast.

File: MapboxNavigation/CarPlayMapViewController.swift [hunk 2]
```diff
@@ -24,7 +26,9 @@ class CarPlayMapViewController: UIViewController, MGLMapViewDelegate {
 
     override func viewDidLoad() {
         super.viewDidLoad()
-
+        
+        styleManager = StyleManager(self)
+        styleManager.styles = [CarPlayDayStyle(), CarPlayNightStyle()]
```
Comment (vincethecoder) [inline MapboxNavigation/CarPlayMapViewController.swift]: @1ec5 should we consider creating a `BaseViewController` that assigns these styles and any other shared configurations for the view controllers? I see some repetition with styles for both `CarPlayMapViewController` and `CarPlayNavigationViewController`.

**Comment by vincethecoder** [comment]
@1ec5 aside the changes for the appearances, everything else looks great! 👍

**Comment by akitchen** [comment]
What's left to be done here?

**Pull request merged**